src/mapeo-project.js

```diff
@@ -631,6 +632,73 @@ export class MapeoProject extends TypedEmitter {
     // 4. Assign LEFT role for device
     await this.#capabilities.assignRole(this.#deviceId, LEFT_ROLE_ID)
   }
+
+  /** @param {Object} opts
+   *  @param {string} opts.configPath
+   */
+  async importConfig({ configPath }) {
+    // check for already present fields and presets and delete them if exist
+    await deleteAll(this.preset)
+    await deleteAll(this.field)
+
+    const config = await readConfig(configPath)
+    /** @type {Map<string, string>} */
+    const iconNameToId = new Map()
+    /** @type {Map<string, string>} */
+    const fieldNameToId = new Map()
+
+    // Do this in serial not parallel to avoid memory issues (avoid keeping all icon buffers in memory)
+    for await (const icon of config.icons()) {
+      const iconId = await this.#iconApi.create(icon)
+      iconNameToId.set(icon.name, iconId)
+    }
+
+    // Ok to create fields and presets in parallel
+    const fieldPromises = []
+    for (const { name, value } of config.fields()) {
+      fieldPromises.push(
+        this.#dataTypes.field.create(value).then(({ docId }) => {
+          fieldNameToId.set(name, docId)
+        })
+      )
+    }
+    await Promise.all(fieldPromises)
+
+    const presetsWithRefs = []
+    for (const { fieldNames, iconName, value } of config.presets()) {
+      const fieldIds = fieldNames.map((fieldName) => {
+        const id = fieldNameToId.get(fieldName)
+        if (!id) {
+          throw new Error(
+            `field ${fieldName} not found (referenced by preset ${value.name})})`
+          )
+        }
+        return id
+      })
+      presetsWithRefs.push({
+        ...value,
+        iconId: iconName && iconNameToId.get(iconName),
+        fieldIds,
+      })
+    }
+    const presetPromises = presetsWithRefs.map((preset) =>
+      this.preset.create(preset)
+    )
+    const createdPresets = await Promise.all(presetPromises)
+    const presetIds = createdPresets.map(({ docId }) => docId)
+
+    await this.$setProjectSettings({
+      defaultPresets: {
+        point: presetIds,
+        line: [],
+        area: [],
+        vertex: [],
+        relation: [],
+      },
+    })
+
+    await config.close()
```

Comment:
What if we did this a little earlier in the function so we could close the file sooner (and avoid the case where some operation, like `$setProjectSettings`, fails and we fail to close the file)?

Reply:
Regarding where to put close, I'm on the fence between putting it after reading the presets file [here](https://github.com/digidem/mapeo-core-next/blob/8e31ab08dccbaeb0a78ff32a0e0867b321fd658c/src/mapeo-project.js#L684) or after creating the presets records [here](https://github.com/digidem/mapeo-core-next/blob/8e31ab08dccbaeb0a78ff32a0e0867b321fd658c/src/mapeo-project.js#L687). 
For now I'm putting it after reading the presets file since that's when we're sure we won't be needing that handle anymore...